src/EventStore.Client/ArrayExtensions.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+
+namespace EventStore.Client {
+	internal static class ArrayExtensions {
+		public static void RandomShuffle<T>(this T[] arr, int i, int j) {
+			if (i >= j)
+				return;
+			var rnd = new Random(Guid.NewGuid().GetHashCode());
+			for (int k = i; k <= j; ++k) {
```

Comment:
@thefringeninja nit pick: we can stop the iteration at `k < j` instead of `k <= j` since there'll be only one element left.

Reply:
Fixed